test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partnership/PartnershipPartnersContactNameControllerSpec.scala

```diff
@@ -0,0 +1,125 @@
+import play.api.http.Status.{BAD_REQUEST, OK}
+import play.api.libs.json.Json
+import play.api.test.DefaultAwaitTimeout
+import play.api.test.Helpers.status
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.DownstreamServiceError
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.group.MemberName
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.partnerships.member_name_page
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+
+class PartnershipPartnersContactNameControllerSpec extends ControllerSpec with DefaultAwaitTimeout {
+
+  private val page = mock[member_name_page]
+  private val mcc  = stubMessagesControllerComponents()
+
+  private val controller =
+    new PartnershipOtherPartnerContactNameController(authenticate = mockAuthAction,
+                                                     journeyAction = mockJourneyAction,
+                                                     registrationConnector =
+                                                       mockRegistrationConnector,
+                                                     mcc = mcc,
+                                                     page = page
+    )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    when(page.apply(any(), any(), any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(page)
+    super.afterEach()
+  }
+
+  private def registrationWithPartnershipDetails =
+    aRegistration(withPartnershipDetails(Some(generalPartnershipDetails)))
+
+  "PartnershipOtherPartnerEmailAddressController" should {
+
+    "return 200" when {
+      "user is authorised, a registration already exists with already collected nominated partner" in {
+        authorizedUser()
+        mockRegistrationFind(registrationWithPartnershipDetails)
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+    }
+
+    "return 400 (BAD_REQUEST)" when {
+      "user does not enter name" in {
+        authorizedUser()
+        val result =
+          controller.submit()(postRequestEncoded(MemberName("John", ""), saveAndContinueFormAction))
+
+        status(result) mustBe BAD_REQUEST
+      }
+
+      "user enters a long name" in {
+        authorizedUser()
+        val result = controller.submit()(
+          postRequestEncoded(MemberName("abced" * 40, "Smith"), saveAndContinueFormAction)
+        )
+
+        status(result) mustBe BAD_REQUEST
+      }
+
+      "user enters non-alphabetic characters" in {
+        authorizedUser()
+        val result =
+          controller.submit()(
+            postRequestEncoded(MemberName("FirstNam807980234£$", "LastName"),
+                               saveAndContinueFormAction
+            )
+          )
+
+        status(result) mustBe BAD_REQUEST
+      }
+    }
+
```

Comment:
I think we should have a test to verify that the registration is updated as expected when form input is valid and that routing is as expected.